src/OpenIddict.Core/OpenIddictExtensions.cs

```diff
@@ -49,9 +49,10 @@ public static class OpenIddictExtensions {
             [NotNull] this OpenIddictBuilder builder,
             [NotNull] string name, int position,
             [NotNull] Action<IApplicationBuilder> registration) {
-            // By default, prevent duplicate registrations.
-            if (builder.Modules.Any(module => string.Equals(module.Name, name))) {
-                return builder;
+            // By default, replace duplicate registrations.
+            var replacedModule = builder.Modules.Where(module => string.Equals(module.Name, name)).ToList();
```

Comment:
Ah yeah, nice catch. I had not realized we were altering the collection in the foreach block :smile: